xdo: from 0.3 -> 0.5

Adjective-Object opened the pull request:
###### Tests done
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- [x] Built on platform(s): NixOS / OSX / Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### Changes made
1. Updated xdo to the most recent major release.
   This incorporates a number of bugfixes, as well as adding the
   `below` and `above` actions.
2. Moved source fetching from `fetchurl` to `fetchFromGitHub`.

cc @meisternu

Adjective-Object committed:
xdo from 0.3 -> 0.5

This incorporates a number of bugfixes, as well as adding the
`below` and `above` actions

Also moved package from fetchurl to fetchFromGitHub

joachifm commented:
What is the reason for building from a git checkout instead of getting the release tarball?

Adjective-Object commented:
When I was trying to fetch the tarball for the version 0.5 release last night, the xdo that was built was still version 0.3. I've just tested it now an building against the the release tarball works fine. In all likelihood, I was probably accidentally building against master at the time.

I switched back to fetchurl and squashed the changes into the original commit. The build products still work, it still builds in chroot, and it still passes nox-review.

joachifm commented:
Thanks

joachifm commented:
@Adjective-Object btw. I was wrong about `fetchFromGitHub`, you were right to use it. Sorry about that.
